dashboard/src/components/Header/ContextSelector.tsx

```diff
@@ -0,0 +1,134 @@
+import { push } from "connected-react-router";
+import React, { useEffect, useRef, useState } from "react";
+import { useDispatch } from "react-redux";
+
+import { angleIcon, ClarityIcons, clusterIcon, fileGroupIcon } from "@clr/core/icon-shapes";
+import { IClustersState } from "../../reducers/cluster";
+import { CdsButton, CdsIcon } from "../Clarity/clarity";
+import useOutsideClick from "../js/hooks/useOutsideClick/useOutsideClick";
+
+import Column from "components/js/Column";
+import { definedNamespaces } from "shared/Namespace";
+import { app } from "shared/url";
+import Row from "../js/Row";
+import "./ContextSelector.css";
+
+ClarityIcons.addIcons(clusterIcon, fileGroupIcon, angleIcon);
+
+export interface IContextSelectorProps {
+  clusters: IClustersState;
+  defaultNamespace: string;
+  fetchNamespaces: () => void;
+  createNamespace: (ns: string) => Promise<boolean>;
+  getNamespace: (ns: string) => void;
+  setNamespace: (ns: string) => void;
+}
+
+function ContextSelector({
+  clusters,
+  defaultNamespace,
+  fetchNamespaces,
+  createNamespace,
+  getNamespace,
+  setNamespace,
+}: IContextSelectorProps) {
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const currentCluster = clusters.clusters[clusters.currentCluster];
+  const namespaceSelected = currentCluster.currentNamespace || defaultNamespace;
+  const [cluster, setCluster] = useState(clusters.currentCluster);
+  const [namespace, setNS] = useState(namespaceSelected);
```

Comment:
I'd be consistent with the naming:

```suggestion
  const [namespace, setNamespace] = useState(namespaceSelected);
```

Reply:
`setNamespace` is already used (by a function of the props)